src/Console/Command/ObfuscateData.php

```diff
@@ -0,0 +1,305 @@
+     * @var string
+     */
+    protected $name = 'db:obfuscatedata';
+
+    /**
+     * The console command signature.
+     *
+     * @var string
+     */
+    protected $signature = 'db:obfuscatedata';
+
+    /**
+     * The console command description.
+     *
+     * @var string
+     */
+    protected $description = 'Obfuscates selected PII and removes various config keys from database';
+
+    public function __construct()
+    {
+        parent::__construct();
+    }
+
+    public function handle()
+    {
+        //set up repos
+        $this->contactRepository = service('repository.contact');
+        $this->postRepository = service('repository.post');
+        $this->configRepository = service('repository.config');
+        $this->userRepository = service('repository.user');
+
+        $this->db = service('kohana.db');
+
+        if ($this->isThisAMultisiteInstall()) {
+            if (!getenv('HOST') || strlen(getenv('HOST')) < 1) {
+                $this->alert("ERROR: A host must be specified for a multisite deployment.");
+                exit;
+            }
+        }
+
+        if ($this->isTestEnvironment() || $this->isStagingEnvironment()) {
+            //confirm acknowledgements
+            $this->alert("WARNING: This script will wipe user, contacts, post author and data source data.");
+            if (!$this->confirm("Do you want to continue?")) {
+                $this->info("Request canceled.\n");
+                exit;
+            }
+            if (!$this->confirm('Do you acknowledge that this data will still contain sensitive data?')) {
+                $this->info("Request canceled.\n");
+                exit;
+            }
+
+            // Do overwriting
+            $this->db->begin();
+            $this->overwriteContacts();
+            $this->overwritePostAuthors();
+            $this->removeDataProviderValues();
+            $this->overwriteSiteConfig();
+            $this->deleteUsers();
+            $this->addAdminUser();
+            $this->db->commit();
+        } else {
+            $this->info("This script will only run on test or staging deployments.\n\n");
+            exit;
+        }
+    }
+
+    protected function isThisAMultisiteInstall()
+    {
+        // @TODO: is this the correct way to check against multisite
+        return (config('multisite.domain'));
+    }
+
+    private function isTestEnvironment()
+    {
+        $cur_env = $this->getLaravel()->environment();
+        return (strtolower($cur_env) == 'test');
+    }
+
+    private function isStagingEnvironment()
+    {
+        $cur_env = $this->getLaravel()->environment();
+        return (strtolower($cur_env) == 'staging');
+    }
+
+    /* optional implementations...
+    private function randomizeLongitudeLatitude($boundaries)
+    {
+    }
+    */
+
+    /* TODO: would this be useful?
+    private function scrambleLongitudeLatitude($distanceFromOrigin = null)
+    {
+    }
+    */
+
+    private function overwriteContacts()
```

Comment:
This method is particularly dense. A bit of spacing/comments would help just for easier reading

Reply:
updated